### ISSUE TITLE
CM Autolinks in spec conflict with GFM Autolinks extension

### ISSUE THREAD

**Issue by dudeofawesome** [github/cmark#81]
CM Autolink examples 592 and 595 show how Autolinks shouldn't match for raw links, such as `< http://foo.bar >` and `http://example.com`. When the GFM Autolinks extension is implemented to spec, those links get matched and transformed in to URLs.

This creates a problem for projects that use the examples listed here as tests for their markdown renders, such as [DartLang's markdown package](https://github.com/dart-lang/markdown/blob/master/tool/gfm_tests.json).

**Comment by kivikakk** [comment]
Indeed; our spec tester only enables the extensions for the examples in the spec that are about that particular extension, see https://github.com/github/cmark/blob/9137851accb62e8ebf23269234b231a1eedc1367/test/spec_tests.py#L107-L115 and https://github.com/github/cmark/blob/9137851accb62e8ebf23269234b231a1eedc1367/test/spec.txt#L8919-L8923.
